parser/prism/Translator.cc

```diff
@@ -1046,7 +1046,53 @@ unique_ptr<parser::Node> Translator::translate(pm_node_t *node, bool preserveCon
 
             auto arg = translate(definedNode->value);
 
-            return make_unique<parser::Defined>(location.join(arg->loc), move(arg));
+            if (!directlyDesugar) {
+                return make_unique<parser::Defined>(location.join(arg->loc), move(arg));
+            }
+
+            // Desugar to `::Magic.defined_instance_var(ivar)` or `::Magic.defined_class_var(cvar)`
+            if (auto *ivar = parser::NodeWithExpr::cast_node<parser::IVar>(arg.get())) {
+                auto sym = MK::Symbol(arg->loc, ivar->name);
```

Comment:
In a follow-up PR, could you please tweak this to only access data from the Prism tree?

That way, when we start the big de-whitequarkification and stop producing the wq tree entirely, this code can continue to work.

It would be something like:

```suggestion
            if (PM_NODE_TYPE(definedNode->value, PM_INSTANCE_VARIABLE_READ_NODE) {
                 auto loc = translateLocation(definedNode->value);
                 auto name = translateConstantName(instanceVarNode->name);
                 auto sym = MK::Symbol(loc, name);
```